### PR TITLE
Add ape_e spectral factor model

### PR DESCRIPTION
**Pull request by RDaxini**
- [X] Relates to #2065 
 - [X] I am familiar with the [contributing guidelines](https://pvlib-python.readthedocs.io/en/latest/contributing.html)
 - [X] Tests added
 - [X] Updates entries in [`docs/sphinx/source/reference`](https://github.com/pvlib/pvlib-python/blob/main/docs/sphinx/source/reference) for API changes.
 - [ ] Adds description and name entries in the appropriate "what's new" file in [`docs/sphinx/source/whatsnew`](https://github.com/pvlib/pvlib-python/tree/main/docs/sphinx/source/whatsnew) for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` or this Pull Request with `` :pull:`num` ``. Includes contributor name and/or GitHub username (link with `` :ghuser:`user` ``).
 - [X] New code is fully documented. Includes [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings, examples, and comments where necessary.
 - [X] Pull request is nearly complete and ready for detailed review.
 - [x] Maintainer: Appropriate GitHub Labels (including `remote-data`) and Milestone are assigned to the Pull Request and linked Issue.

This function calculates the spectral mismatch factor using the average photon energy and the depth of a water absorption band as inputs. Context: #2065

### CODE REVIEW

File: pvlib/spectrum/mismatch.py [hunk 1]
```diff
@@ -836,6 +836,132 @@ def spectral_factor_pvspec(airmass_absolute, clearsky_index,
...
+
+        * ``asi-t`` - anyonymous triple junction amorphous Si module
+        * ``'cdte'`` - anyonymous CdTe module.
+        * ``'multisi'`` - anyonymous multicrystalline Si module
+
+        The default is None.
+
+    coefficients : array-like, optional
+        User-defined coefficients, if not using one of the default coefficient
+        sets via the ``module_type`` parameter. The default is None.
```
Comment (echedey-ls) [inline pvlib/spectrum/mismatch.py:865]: Can we remove "The default is None."?

#1574 , #1828, #2084

File: pvlib/spectrum/mismatch.py [hunk 2]
```diff
@@ -836,6 +836,132 @@ def spectral_factor_pvspec(airmass_absolute, clearsky_index,
...
+    photon energy (APE) and the depth of a water absorption band using the
+    Daxini model.
+
+    Parameters
+    ----------
+    ape : numeric
+        Average photon energy (APE, :math:`\varphi`) [eV]
+
+    band_depth : numeric
+        Depth of a spectral band, :math:`\varepsilon` [Wm:math:`^{-2}`]
```
Comment (echedey-ls) [inline pvlib/spectrum/mismatch.py:852]:
```suggestion
        Depth of a spectral band, :math:`\varepsilon` [W/m²]
```

File: pvlib/tests/test_spectrum.py [hunk 3]
```diff
@@ -474,6 +474,43 @@ def test_spectral_factor_jrc_supplied_ambiguous():
...
+    ('cdte', pd.Series([0.94950, 0.98647, 1.04016])),
+    ('asi-t', pd.Series([0.92752, 1.02894, 1.13527])),
+])
+def test_spectral_factor_daxini_series(module_type, expected):
+    apes = pd.Series([1.82, 1.87, 1.95])
+    bands = pd.Series([2, 4, 8])
+    out = spectrum.spectral_factor_daxini(apes, bands,
+                                          module_type=module_type)
+    assert isinstance(out, pd.Series)
+    assert np.allclose(expected, out, atol=1e-5)
```
Comment (echedey-ls) [inline pvlib/tests/test_spectrum.py:488]:
```suggestion
    assert_allclose(out, expected, atol=1e-5)
```

**Review by echedey-ls — COMMENTED** [review]
Nice job with this implementation!

I miss a test of numpy input/output datatypes. Also, sphinx warns something near line 64 of the documentation, but I can't grasp what it could be.

**Comment by RDaxini** [comment]
@echedey-ls Thanks for the review.
I am waiting for a conclusion to the discussion in #2126 before working on (or closing...) this PR.

**Comment by RDaxini** [comment]
Closed for now, see discussion in #2065 regarding availability of coefficients for more details.

**Comment by RDaxini** [comment]
Closed for now, see discussion in https://github.com/pvlib/pvlib-python/issues/2065 regarding availability of coefficients for more details.

**Pull request closed without merging**